gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/CertificateServiceProxyImpl.java

```diff
@@ -31,60 +35,128 @@
+        return Stream.of(expiryThresholds.trim().split(","))
+                .map(String::trim)
+                .map(Integer::valueOf)
+                .sorted(Comparator.reverseOrder())
+                .toList()
+                .get(0);
+    }
 
     @Override
     public Maybe<Certificate> findById(String id) {
         return certificateService.findById(id).flatMap(reporter -> filterSensitiveData(reporter).toMaybe());
     }
 
     @Override
-    public Flowable<Certificate> findAll() {
-        return certificateService.findAll().flatMapSingle(this::filterSensitiveData);
+    public Single<List<CertificateEntity>> findByDomainAndUse(String domainId, String use) {
+        return certificateService.findByDomain(domainId)
+                .filter(cert -> {
+                    if (StringUtils.isBlank(use)) {
+                        return true;
+                    }
+                    return cert.hasUse(use, objectMapper::readTree);
+                })
+                .flatMapSingle(this::filterSensitiveData)
+                .toList()
+                .map(allCerts -> {
+                    var renewedSystemCertIds = allCerts
+                            .stream()
+                            .filter(Certificate::isSystem)
+                            .sorted(new CertificateTimeComparator())
+                            // all system certs are renewed - except for the newest one
+                            .skip(1)
+                            .map(Certificate::getId)
+                            .collect(Collectors.toSet());
+                    return allCerts
+                            .stream()
+                            .map(c -> Pair.of(c, renewedSystemCertIds.contains(c.getId())))
+                            .toList();
+                })
+                .flattenAsFlowable(certsAndIds -> certsAndIds)
+                .flatMapSingle(pair -> {
+                    var cert = pair.getLeft();
+                    var isRenewedSystemCert = pair.getRight();
+                    return Single.zip(
+                            getAppsUsing(cert).toList(),
+                            getIdpsUsing(cert).toList(),
+                            (appUsage, idpUsage) -> CertificateEntity.forList(cert, certExpirationWarningThreshold, isRenewedSystemCert, appUsage, idpUsage)
+                    );
+                })
+                .sorted(Comparator.comparing(CertificateEntity::name, String.CASE_INSENSITIVE_ORDER))
+                .toList();
     }
 
-    @Override
-    public Flowable<Certificate> findByDomain(String domain) {
-        return certificateService.findByDomain(domain).flatMapSingle(this::filterSensitiveData);
+    private Flowable<IdentityProvider> getIdpsUsing(Certificate cert) {
+        return idps.findByCertificate(cert.getDomain(), cert.getId())
```

Comment:
**Quetion**: why are you doing the filtering into the repository layer instead of in this servicez class? With you implementation you are duplicating the code for both Mong/RDBMS where it could be manage in a single place here.